src/toHaveStyleRule.js

```diff
@@ -20,10 +20,18 @@ const getClassNames = received => {
 const hasClassNames = (classNames, selectors) =>
   classNames.some(className => selectors.includes(`.${className}`))
 
-const getRules = (ast, classNames) =>
-  ast.stylesheet.rules.filter(
+const getRules = (ast, classNames, options) => {
+  let rules = ast.stylesheet.rules
+  if (options.media) {
+    rules = rules
```

Comment:
Yep, both looks good 👍 . I've been thinking in this way before too. If it could dynamically use the options key for filtering rules. But at first i wasn't sure if handling others than ```@media``` are the same. If number 2 is apply, is number 1 gonna look like this?
```js
const rules = Object.keys(options).length ? getAtRules(ast, options.media) : ast.stylesheet.rules
```